### ISSUE TITLE
Dataverse Header/Theme - New layout, default colors

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#4197]
Changes to the default header layout with gray background at the top of page should be more minimal, removing some of the spacing/padding as well as the background color (or changing it to "white").

From a comment by @TaniaSchlatter on #4190

<img width="1045" alt="custom_dataverse_page" src="https://user-images.githubusercontent.com/15386424/31507438-afb30318-af48-11e7-9685-e76e1ab14eea.png">
<img width="1058" alt="custom_newdataverse" src="https://user-images.githubusercontent.com/15386424/31507458-bf6d05e2-af48-11e7-83e9-1773df325b9b.png">

This would not affect themes that have already been created:
![custom_journal](https://user-images.githubusercontent.com/15386424/31507469-c6aa9978-af48-11e7-98a2-90cb257ac8d1.jpg)

An additional change will be to disable the dataverse theme header for the root dataverse. If they choose to utilize the new customization features, which also for easier branding in the navbar and the space above it, then it lessens the need for the theme header directly below the navbar. Here is a screenshot for how it would look on a dataset pg that belongs to the Harvard Dataverse.

<img width="1039" alt="custom_harvard_dataset" src="https://user-images.githubusercontent.com/687227/32065420-c15ae790-ba4a-11e7-8401-d22fcd39184d.png">

**Comment by pdurbin** [comment]
I just pushed 7795e70 to kick off a new branch, changing the default header background from gray to white.

**Comment by pdurbin** [comment]
In 8eccacd I made it so only non-root dataverses show the header.

I went ahead and made pull request #4277 but I have questions.

If we aren't showing the header for a root dataverse, should we hide the "Inherit customization from Root" checkbox from dataverses that are direct children of the root dataverse? Here's a screenshot of the "Birds" dataverse, whose parent dataverse is the "Root" dataverse:

![screen shot 2017-11-13 at 3 57 21 pm](https://user-images.githubusercontent.com/21006/32748839-6948aed0-c88b-11e7-9fbc-a1cd16cd59d4.png)

**Comment by mheppler** [comment]
The disabled theme for the root dataverse should be an option added to the Theme + Widget pg for the root, not a new default setting. Moving back to develop.

**Comment by pdurbin** [comment]
> The disabled theme for the root dataverse should be an option added to the Theme + Widget pg for the root

Oh, I didn't realize we were building a UI for this. (Does anyone remember how this got estimated as a 2?) It would be easier to add a database setting called `:DisableRootDataverseTheme` if that's ok. Assuming we are building a UI, I could use some design input. For now I threw a "disable theme" checkbox on the top of the page like this:

![screen shot 2017-11-15 at 8 53 22 am](https://user-images.githubusercontent.com/21006/32839586-e6d9a900-c9e2-11e7-83c6-08f668314fae.png)

Again, I would prefer to not build a UI at this time and implement a database setting instead. Let's talk about it during or after standup.

**Comment by scolapasta** [comment]
Let's remember our general guideline that we want to consider the root dataverse the same as any other dataverse as much as possible.

With that guideline in mind, I agree with @pdurbin it makes more sense for this to be a setting.

If we want this to be an option for any dataverse (I could imagine any dataverse might want to streamline and not have a block header), then it should be in the UI. But that seems potentially out of scope for this issue (i.e. we could make it a setting now, and make it a db column for any dataverse for a future issue)

**Comment by pdurbin** [comment]
@mheppler @scolapasta @TaniaSchlatter and I met after standup and made some decisions. Here's a todo list.

- [ ] Implement `:DisableRootDataverseTheme` (or some such name) as a database setting rather than a UI checkbox.
- [ ] When the root dataverse theme is disabled, render some explanatory text on the "Theme" tab with a "Done" button that takes you back, just like how the "Done" button works on the "Widgets" tab.
- [ ] When the root dataverse theme is disabled, don't show it on the home/dataverse page when you're looking at the root dataverse.

Not discussed, but implied:

- [ ] When the root dataverse theme is disabled, do not offer the "Inherit customization from Root" checkbox from dataverses that are direct children of the root dataverse.

Thanks for getting me unblocked, all.

**Comment by pdurbin** [comment]
I'm not sure if we want to disable that "Inherit customization from Root" checkbox or not (I took a quick look but didn't want to wast time) but I implemented the items we discussed this morning above in 28f705c.

Moving to code review. @mheppler you might want to pull the `4197-dv-header` branch to make sure this is working the way we want (please note the setting and curl command I added). @dlmurphy there's an opportunity for wordsmithing here.

Some screnshots:

## Explanatory text that the root dataverse theme has been disabled

![screen shot 2017-11-15 at 12 51 16 pm](https://user-images.githubusercontent.com/21006/32852268-5edd0ccc-ca05-11e7-8989-2d0dd8d1de8f.png)

## No theme on root dataverse

![screen shot 2017-11-15 at 12 51 22 pm](https://user-images.githubusercontent.com/21006/32852266-5ecf8890-ca05-11e7-9de2-a49a23948dd0.png)

## A theme remains on non-root dataverses

![screen shot 2017-11-15 at 12 51 33 pm](https://user-images.githubusercontent.com/21006/32852265-5ebd7466-ca05-11e7-8322-c05bf1780b49.png)

**Comment by dlmurphy** [comment]
Forgot to tag it with the issue number, but I just committed a small fix to the docs in 80219c5. Going to hold off on further doc review until @mheppler and @pdurbin investigate the stuff we discussed this morning.

**Comment by pdurbin** [comment]
> the "Inherit customization from Root" checkbox

@mheppler and I discussed this after standup and he's going to pull the branch and help us decide what direction we should take.

We also talked with @dlmurphy about how I could implement an admin API endpoint to check if any dataverses that are direct children of the root dataverse are inheriting its theme. Stay tuned.

**Comment by mheppler** [comment]
Reviewed options for inherit with @pdurbin and @scolapasta. Added tip to Installation Guide > Configuration > Custom Header with suggestions for how to handle child dataverses that inherit themes from root. @dlmurphy please review to make sure it is sufficiently legible. @matthew-a-dunlap continue with your code review.

**Comment by pdurbin** [comment]
GitHub doesn't allow me to approve a pull request that I created but I approve of the subsequent changes @mheppler made. Looks good.

@matthew-a-dunlap are you still planning on doing some code review? I'm fine with this issue moving to QA if there are no objections.

**Comment by dlmurphy** [comment]
Docs look good to me. Letting this stay in code review in case @matthew-a-dunlap is reviewing it.

**Comment by pdurbin** [comment]
Moving to QA after @matthew-a-dunlap said, "feel free to move forward on 4197 without my review!"

**Comment by matthew-a-dunlap** [comment]
I've reviewed it, looks good, and I will move it for real!

**Comment by kcondon** [comment]
-Gray background for root dataverse still present in default customization, works for all sub dataverses.

**Comment by pdurbin** [comment]
I fixed the gray background bug in 624922f

@mheppler @kcondon and I have been talking about this issue all morning.

I'm fine with the code as is (unless more bugs are found) but @mheppler was thinking about looking at how `FFFFFF` (white) is now hard coded on the front end and @dlmurphy may want to look at what's written in the guides about inheriting themes because @kcondon and I agree it's a bit confusing. There's a drawing on my whiteboard that might help.

**Comment by mheppler** [comment]
Worked with @dlmurphy @pdurbin and @kcondon to get the documentation more explicit when describing how the disable config option impacts inherited themes. That should clear the way for this to get through QA.

Also, noticed that the Theme + Widgets pg referred to the root's "customization" for the inherit feature, so I went ahead and corrected that to refer to the root's "theme" instead.
